pkg/api/v1/summary_test.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2023 The TestGrid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"testing"
+
+	apipb "github.com/GoogleCloudPlatform/testgrid/pb/api/v1"
+	configpb "github.com/GoogleCloudPlatform/testgrid/pb/config"
+	summarypb "github.com/GoogleCloudPlatform/testgrid/pb/summary"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/testing/protocmp"
+)
+
+func TestListTabSummaries(t *testing.T) {
+	tests := []struct {
+		name        string
+		config      map[string]*configpb.Configuration
+		summaries   map[string]*summarypb.DashboardSummary
+		req         *apipb.ListTabSummariesRequest
+		want        *apipb.ListTabSummariesResponse
+		expectError bool
+	}{
+		{
+			name: "Returns an error when there's no dashboard in config",
+			config: map[string]*configpb.Configuration{
+				"gs://default/config": {},
+			},
+			req: &apipb.ListTabSummariesRequest{
+				Dashboard: "missing",
+			},
+			expectError: true,
+		},
+		{
+			name: "Returns an error when there's no summary for dashboard yet",
+			config: map[string]*configpb.Configuration{
+				"gs://default/config": {
+					Dashboards: []*configpb.Dashboard{
+						{
+							Name: "ACME",
+							DashboardTab: []*configpb.DashboardTab{
+								{
+									Name:          "me-me",
+									TestGroupName: "testgroupname",
+								},
+							},
+						},
+					},
+				},
+			},
+			summaries: map[string]*summarypb.DashboardSummary{
+				"gs://default/summary/summary-testgroup1": {},
```

Comment:
This dashboard name doesn't match the config, and shouldn't be read from, right?

Please either do not include it or name it "summary-unrelateddashboard" or whatever if that's what you're meaning.

Reply:
Addressed this above. Michelle also pointed it out - it's empty now.